vignettes/Workflow.Rmd

```diff
@@ -0,0 +1,479 @@
+score cutoff was also applied. The remaining data were reduced to a set
+of aligned intervals (including orientation). This saves a great deal of
+memory, as the sequences, which are unnecessary, are discarded. Finally,
+we subset the data for chr10 to chr12, simply for convenience in this
+vignette.
+
+We outline this process with this unevaluated code block:
+```{r preprocess, eval=FALSE}
+ qa_list <- lapply(sampleFiles, qa)
+report(do.call(rbind, qa_list)) 
+## spend some time evaluating the QA report, then procede 
+filter <- compose(chipseqFilter(),
+alignQualityFilter(15)) 
+cstest <- GenomicRangesList(lapply(sampleFiles, function(file) 
+as(readAligned(file, filter), "GRanges") )) 
+cstest <-cstest[seqnames(cstest) in c("chr10", "chr11", "chr12")]
+```
+
+The above step has been performed in advance,
+and the output has been included as a dataset in this package. We load
+it now: 
+```{r} 
+data(cstest)
+cstest 
+```
+
+```{r convert-cstest, echo=FALSE, eval=FALSE}
+## code used to convert the GenomeDataList to a
+GRangesList cstest <- GenomicRangesList(lapply(cstest, function(gd)
+  do.call(c, lapply(names(gd), function(chr) 
+    pos <- gd[[chr]] 
+    starts <- c( pos[["-"]] - 23L, pos[["+"]]) 
+    GRanges(chr, IRanges(starts, width = 24), 
+           rep(c("-", "+"), elementNROWS(pos))) ))
+)) 
+```
+
+`cstest` is an object of class *GRangesList*, and has a list-like
+structure, each component representing the alignments from one lane, as
+a *GRanges* object of stranded intervals. 
+```{r}
+cstest$ctcf
+```
+
+
+#Extending reads
+
+
+Solexa gives us the first few (24 in this example) bases of each
+fragment it sequences, but the actual fragment is longer.  By design,
+the sites of interest (transcription factor binding sites) should be
+somewhere in the fragment, but not necessarily in its initial part.
+Although the actual lengths of fragments vary, extending the alignment
+of the short read by a fixed amount in the appropriate direction,
+depending on whether the alignment was to the positive or negative
+strand, makes it more likely that we cover the actual site of
+interest.
+
+
+It is possible to estimate the fragment length, through a variety of
+methods. There are several implemented by the
+`estimate.mean.fraglen` function. Generally, this only needs
+to be done for one sample from each experimental protocol. Here, we
+use SSISR, the default method: 
+```{r estimate.mean.fraglen}
+fraglen <- estimate.mean.fraglen(cstest$ctcf, method="correlation")
+fraglen[!is.na(fraglen)] 
+```
+
+Given the suggestion of $~190$ nucleotides,
+we extend all reads to be 200 bases long. This is done using the
+`resize` function, which considers the strand to determine the direction
+of extension: 
+```{r}
+ctcf.ext <- resize (cstest$ctcf, width = 200)
+ctcf.ext
+```
+
+
+We now have intervals for the CTCF lane that represent the original
+fragments that were precipitated.
+
+#Coverage, islands, and depth
```

Comment:
Put a space between `#` and `Coverage`. Do this for other instances in this vignette.